index.d.ts

```diff
@@ -0,0 +1,236 @@
+ * It will use doc1 if it fits entirely on the current line, otherwise it will use doc2.
+ * More precisely, doc1 will be used iff:
+ *   1. It can be rendered flat. A "flat" document has no newlines, i.e., no vert. And,
+ *   2. When rendered flat, it fits on the current line without going over the
+ *      pretty printing width.
+ *
+ * @param doc1 a document
+ * @param doc2 another document
+ */
+export function ifFlat(doc1: DocLike, doc2: DocLike): Doc;
+
+/**
+ * ensures that nothing is placed after doc, if at all possible.
+ * @param doc a document
+ */
+export function fullLine(doc: DocLike): Doc;
+
+/**
+ * `sepBy(items, sep, vertSep="")` will display either:
+ *
+ *     items[0] sep items[1] sep ... items[n]
+ *
+ * if it fits on one line, or:
+ *
+ *     items[0] vertSep \n items[1] vertSep \n ... items[n]
+ *
+ * otherwise. (Without the extra spaces; those are there for readability.)
+ *
+ * Neither `sep` nor `vertSep` may contain newlines.
+ * @param items list of documents
+ * @param sep separator string
+ * @param vertSep vertical separator string
+ */
+export function sepBy(items: DocLike[], sep?: string, vertSep?: string): Doc;
+
+/**
+ * `wrap(words, sep=" ", vertSep="")` does word wrapping. It combines the `words` with
+ * `sep` when they fit on the same line, or `vertSep\n` when they don't.
+ *
+ * For simple word wrapping, you would use:
+ *
+ *     wrap(words, " ", "") // or just wrap(words)
+ *
+ * For word-wrapping a comma-separated list, you would use:
+ *
+ *     wrap(words, ", ", ",")
+ *
+ * Neither `sep` nor `vertSep` may contain newlines.
+ */
+export function wrap(words: DocLike[], sep?: string, vertSep?: string): Doc;
+
+/**
+ * `standardSexpr(func, args)` is rendered like this:
+ *
+ *      (func args ... args)
+ *
+ * or like this:
+ *
+ *      (func
+ *       args
+ *       ...
+ *       args)
+ */
+export function standardSexpr(func: DocLike, args: DocLike[]): Doc;
+
+/**
+ * `lambdaLikeSexpr(keyword, defn, body)` is rendered like this:
+ *
+ *     (keyword defn body)
+ *
+ * or like this:
+ *
+ *     (keyword defn
+ *       body)
+ */
+export function lambdaLikeSexpr(
+  keyword: DocLike,
+  defn: DocLike,
+  body: DocLike
+): Doc;
+
+/**
+ * `beginLikeSexpr(keyword, bodies)` is rendered like this:
+ *
+ *     (keyword
+ *       bodies
+ *       ...
+ *       bodies)
+ */
+export function beginLikeSexpr(keyword: DocLike, bodies: DocLike[]): Doc;
+
+/**
+ * shorthand for building a `doc`, called `pretty`. Itaccepts template strings that
+ * may contain newlines. It combines the lines with `vert`, and the parts of each
+ * line with `horz`. For example, this template:
+ *
+ * ```js
+ *     pretty`if (${c}) {\n  ${t}\n} else {\n  ${e}\n}`)
+ * ```
+ *
+ * pretty prints an `if` statement across multiple lines:
+ *
+ *     if (a == b) {
+ *       a << 2
+ *     } else {
+ *       a + b
+ *     }
+ */
+export function pretty(strs: string[], ...vals: DocLike[]): string;
```

Comment:
It wasn't obvious to me from the documentation/code whether pretty() is supposed to return a string or something else...

Reply:
cc @schanzer